python/cudf/cudf/core/dataframe.py

```diff
@@ -2271,12 +2204,18 @@ def reindex(
         Parameters
         ----------
         labels : Index, Series-convertible, optional, default None
-        axis : {0 or 'index', 1 or 'columns'}, optional, default 0
         index : Index, Series-convertible, optional, default None
             Shorthand for ``df.reindex(labels=index_labels, axis=0)``
         columns : array-like, optional, default None
             Shorthand for ``df.reindex(labels=column_names, axis=1)``
```

Comment:
What are `index_labels` and `column_names`? I think it's better to give concrete description compared to the `shorthand`.

Reply:
Thanks for pointing this out. I made a couple of updates to this docstring:

[pandas docs for reference](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.reindex.html)

- In newer versions of pandas, the kwargs `index`, `columns`, `labels`, and `axis` are not individually documented despite being part of the function signature in the docs. Instead they are grouped under something called `keywords for axes`. This threw me off a bit at first and I wondered if we might want to keep them separate in cuDF. Instead I pulled the parameter descriptions from the older [pandas 1.0 docs](https://pandas.pydata.org/pandas-docs/version/1.0.0/reference/api/pandas.DataFrame.reindex.html?highlight=reindex#pandas.DataFrame.reindex), which broke out most of the parameters. That said, I did choose to break out `columns` and `index` by themselves and wrote my own description for those kwargs. This is all just what makes sense to me most, so happy to make changes here.
- I moved the note about the calling conventions down to the examples section
- I updated the cuDF examples to be the same as the pandas ones and added the one about `fill_value`. 
- I omitted the example that fills a numeric column with a string and casts it to string. This behavior is not supported yet. I am looking into making it happen inside of this PR but can also leave it as a follow up.